src/Controller/ErrorController.php

```diff
@@ -42,6 +43,8 @@ public function initialize(): void
      */
     public function beforeRender(EventInterface $event)
     {
-        $this->viewBuilder()->setTemplatePath('Error');
+        if (Configure::read('debug')) {
```

Comment:
The templates for dev errors like `missing_controller.ctp` reside in non prefixed folder hence when debug is on we always use non prefix error folder.

Reply:
Could this interact poorly with custom exceptions?

Reply:
Depends on from which folder people except the template to be used for those custom exceptions.

Which is why I think it might be better to move the template folder selection inside exception renderer itself as I have mentioned below. If template is error400 or error500 use prefixed folder (if it's a prefixed route) else not.